src/core/src/dimension.cpp

```diff
@@ -41,6 +41,24 @@ Dimension Dimension::operator-(const Dimension& dim) const {
     return Dimension(m_dimension - dim.m_dimension);
 }
 
+Dimension Dimension::operator/(const value_type divisor) const {
+    OPENVINO_ASSERT(divisor >= 0, "divisor must be greater than 0");
+
+    if (m_dimension.get_max_val() == Interval::s_max && m_dimension.get_min_val() == 0)
+        return Dimension::dynamic();
+
+    if (m_dimension.get_max_val() / divisor == m_dimension.get_min_val() / divisor &&
+        m_dimension.get_min_val() > (m_dimension.get_min_val() / divisor * divisor))
+        OPENVINO_UNREACHABLE("Dimension value: [ ",
+                             m_dimension.get_min_val(),
+                             ", ",
+                             m_dimension.get_max_val(),
+                             "]",
+                             " must be a multiple of divisor: ",
+                             divisor);
+    return Dimension(m_dimension.get_min_val() / divisor, m_dimension.get_max_val() / divisor);
+}
```

Comment:
Ok, I see. But I especially would like to get rid of `OPENVINO_UNREACABLE` in here.

You see, this is not enough information to put in the error message and its logic should be handled per operation. Also, to build up some logic and graceful error messages we would have to repeat the same if-statement in our code which doesn't seem useful.

Also, users of `operator/` may need no resulting Dimension but just need to check if the division gone well and for that they call `/` and then check if resulting Dimension is not an empty range. So I propose to go like this

```suggestion
Dimension Dimension::operator/(const value_type divisor) const {
    OPENVINO_ASSERT(divisor >= 0, "divisor must be greater than 0");
    if (m_dimension.get_max_val() == Interval::s_max && m_dimension.get_min_val() == 0)
        return Dimension::dynamic();
    return Dimension(ceil(m_dimension.get_min_val() / divisor),
                    floor(m_dimension.get_max_val() / divisor));
}
```
And during the usage of it always check if output Dimension isn't empty.

Reply:
@jane-intel , I understood your ideas. Since we are using the generic code for Dimension and StaticDimension, there is no  API to check empty for StaticShape.  Also Dimension needs to get_interval().empty() and check but how to check this for StaticDimension?  

Or did you mean StaticDimension still use  OPENVINO_UNREACHABLE but Dimension would not. But it seems OP unified code needs to use template code. Also considering there would be substring check in ov_core_unit_test, unified substring format would also needed.  

 Could you give me some suggestion on it? Thx!